vita3k/gui/src/controls_dialog.cpp

```diff
@@ -43,19 +43,68 @@ static char const *SDL_key_to_string[]{ "[unset]", "[unknown]", "[unknown]", "[u
     "Keypad Mem+", "Keypad Mem-", "Keypad Mem*", "Keypad Mem/", "Keypad +/-", "Keypad Clear", "Keypad ClearEntry", "Keypad Binary", "Keypad Octal",
     "Keypad Dec", "Keypad HexaDec", "[unset]", "[unset]", "LCtrl", "LShift", "LAlt", "Win/Cmd", "RCtrl", "RShift", "RAlt", "RWin/Cmd" };
 
+static const short total_key_entries = 25;
+
+static bool exists_in_array(int *ptr, int val, size_t size) {
+    if (!ptr || !size) {
+        return false;
+    }
+    int *end = ptr + size;
+    return std::find(ptr, end, val) != end;
+}
+
+static void prepare_map_array(EmuEnvState &emuenv, std::array<int, total_key_entries> &map) {
+    map[0] = emuenv.cfg.keyboard_leftstick_up;
```

Comment:
Since this is ordered I would maybe expect this instead. Pretty sure std::array copy constructor is defined nicely here. Alternatively, why don't you return std::array here?
```cpp
std::array entries = {
  emuenv.cfg.keyboard_*,
  ...
}

map = entries;
```